github/pkg/reconciler/githubsource.go

```diff
@@ -173,8 +173,9 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, source *sourcesv1alpha1.G
 		if err != nil {
 			source.Status.MarkNoSecrets("AccessTokenNotFound", "%s", err)
```

Comment:
I would make sure that the error is "NotFound" and do not retry if that the case. Here an example: https://github.com/knative/eventing/blob/master/pkg/reconciler/apiserversource/apiserversource.go#L172

Reply:
Thanks! I have used apierrors.IsNotFound().